### PR TITLE
Issue #696 Added admin user with cluster-admin role

### PR DESCRIPTION
**Pull request by praveenkumar**
This patch will fix cluster-admin addon to provide cluster-admin access to admin user.

### CODE REVIEW

File: addons/cluster-admin/cluster-admin.addon [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Name: cluster-admin
 # Description: Adds the cluster-admin role to the developer user.
 
-oc adm policy add-role-to-user cluster-admin developer
+oc create user admin --full-name=admin
```
Comment (hferentschik) [inline addons/cluster-admin/cluster-admin.addon]: that was easy :-) Good stuff.

File: addons/cluster-admin/cluster-admin.addon [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # Name: cluster-admin
 # Description: Adds the cluster-admin role to the developer user.
```
Comment (hferentschik) [inline addons/cluster-admin/cluster-admin.addon]: The description needs to be changed

**Comment by hferentschik** [comment]
So the idea is to modify our existing default cluster-admin plugin? No new/additional add-on? 

We might want to at least change the name in the addon file to create-cluster-admin. Maybe even rename file and directory name. Thoughts?

@budhrg @LalatenduMohanty ^^

File: addons/cluster-admin/cluster-admin.addon [hunk 3]
```diff
@@ -1,4 +1,5 @@
 # Name: cluster-admin
-# Description: Adds the cluster-admin role to the developer user.
+# Description: Create admin user and add cluster-admin role to it.
 
-oc adm policy add-role-to-user cluster-admin developer
+oc create user admin --full-name=admin
```
Comment (LalatenduMohanty) [inline addons/cluster-admin/cluster-admin.addon]: I think we should go for an user with name "cluster-admin"

File: addons/cluster-admin/cluster-admin.addon [hunk 4]
```diff
@@ -1,4 +1,5 @@
 # Name: cluster-admin
```
Comment (LalatenduMohanty) [inline addons/cluster-admin/cluster-admin.addon]: Also as discussed with @hferentschik @budhrg  @praveenkumar  The addon name should be "create-cluster-addon"

Comment (hferentschik) [inline addons/cluster-admin/cluster-admin.addon]: I assume you mean "create-cluster-admin" ;-)

**Comment by praveenkumar** [comment]
@hferentschik @LalatenduMohanty finally I am endup doing change from directory level as well because when default addons install it go to that directory which named as `addon` name before install https://github.com/minishift/minishift/blob/master/cmd/minishift/cmd/addon/addon_install.go#L84

File: addons/create-cluster-admin/create-cluster-admin.addon [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Name: create-cluster-admin
+# Description: Create admin user and add cluster-admin role to it.
+
+oc create user admin --full-name=admin
```
Comment (LalatenduMohanty) [inline addons/create-cluster-admin/create-cluster-admin.addon]: @praveenkumar I suggest the user name should be "cluster-admin". I think thats what others agreed also @hferentschik @budhrg

Comment (praveenkumar) [inline addons/create-cluster-admin/create-cluster-admin.addon]: @LalatenduMohanty No we agreed about only the addon name not about `user` and why you think user should be `cluster-admin` ?

Comment (hferentschik) [inline addons/create-cluster-admin/create-cluster-admin.addon]: I prefer 'admin', mainly it is easier to type and it aligns with what we had before.

**Comment by jorgemoralespou** [comment]
I think this should be add-cluster-role-to-user since what you're assigning is a cluster wide role.

In case of doubt, better to ask @liggitt.

Jordan,
To create an admin user as cluster admin what command should be?

````
oc adm policy add-role-to-user cluster-admin admin
````

or 

````
oc adm policy add-cluster-role-to-user cluster-admin
````

**Comment by liggitt** [comment]
add-cluster-role-to-user

**Comment by LalatenduMohanty** [comment]
@praveenkumar @hferentschik I suggested `cluster-admin` because It might confuse user with the `system:admin` which comes from `cluster up` output.  We need to educate users that they can not login as system:admin but we have created an alternative for them.  WDYT ?

@jorgemoralespou suggestion?

```
 To login as administrator:
       oc login -u system:admin
```

File: addons/create-cluster-admin/create-cluster-admin.addon [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+oc adm policy add-cluster-role-to-user cluster-admin admin
```
Comment (hferentschik) [inline addons/create-cluster-admin/create-cluster-admin.addon]: btw, there is a new line missing here

Comment (praveenkumar) [inline addons/create-cluster-admin/create-cluster-admin.addon]: @hferentschik will add that.

**Comment by praveenkumar** [comment]
> I suggested cluster-admin but not just admin because otherwise it might confuse user with the system:admin which comes from cluster up output. 

I am not sure how creating cluster-admin as user going to help that. Will do final update on this PR once we all agree what actually we need. (I don't have any strong feeling about username also but I just not convince that changing that might help).

**Comment by hferentschik** [comment]
Ok, so there are several aspects here and I am not sure whether we are all on the same level. 
So what we want to do is to create a new user _admin_ (or _cluster-admin_ as @LalatenduMohanty prefers) who has the cluster admin role. The main use case is to be able to login as this user via the web console and getting access to the otherwise hidden namespaces of OpenShift.

Looking at the actual commands, afaict we are doing the right thing. What we are discussing right now is the metadata of the add-on:

    # Name: create-cluster-admin
    # Description: Create admin user and add cluster-admin role to it.

I think the description is ok, right? Maybe we can be a bit more precise? "Creating a user 'admin' and assigning the cluster-admin role to him"

 Leaves the 'Name', which is what a user sees when he does things like _minishift addons [list|enbable|disable]_. At the moment it is _create-cluster-admin_. I can see how this is ambiguous, on the other hand, things like _create-admin-user-with-cluster-role_ might be a bit unwieldy to mange. As said, this are the names a user needs to work with from the CLI. So  _create-admin-user-with-cluster-role_ is probably a more accurate name, but I am not sure whether it is a good choice. Alternatively, just _admin-user_. It does not explain in the title what it does, but it is less ambiguous and it is the description which adds additional context.

So right now my preferred options are:
* _create-admin-user-with-cluster-role_
* _admin-user_

Can we align on one of these names?

**Comment by LalatenduMohanty** [comment]
> So right now my preferred options are:
> 
>     create-admin-user-with-cluster-role
>     admin-user

As discussed  we will keep the addon name as `admin-user` and the user name as `admin`.

**Pull request merged**